insights/core/plugins.py

```diff
@@ -81,11 +87,33 @@ class datasource(PluginType):
     Decorates a component that one or more :class:`insights.core.Parser`
     subclasses will consume.
     """
+    filterable = False
     multi_output = False
     raw = False
-    filterable = False
+
+    def _get_spec_name(self):
```

Comment:
I think it would be better to modify this function as an `inner` function of the `invoke` and change it to a `recursive` function to get its final `insights.specs.Specs` spec.

Reply:
Ahhh I did forget the recursive part sorry about that. Updated, and moved it to the dr module as it is needed to be called in the serde module during hydration.